### PR TITLE
Feat: Defined rules for `method-specific-id`

### PR DESCRIPTION
**Pull request by arnabghose997**
The PR aims to solidify the standard for `method-specific-id` formats.

The supported formats are:

- **CAIP-10 Blockchain Account ID**: 
  - Example: `did:hid:testnet:cosmos:osmosis-1:osmo1f6r0x3pljpl7pe76zzv36l0ksztqmdltakhv4r`, `did:hid:testnet:eip155:1:0xF4eE129BEDE6ac5E870bCf972e74A117b4809df9`
  - While Registering a DID Document with this `method-specific-id` format, a verification method corresponding to the `method-specific-id`'s blockchain address MUST be provided to prove the ownership.

- **String ID consisting of alphanumeric, dot(.) and hyphen (-) characters**
  - This is done to accommodate ID formats such as UUID, ENS like domans, etc.
  - Unlike former format type, this does not require any ownership proof

### CODE REVIEW

File: x/ssi/types/utils.go [hunk 1]
```diff
@@ -46,3 +47,84 @@ func FindInSlice(list []string, element string) bool {
...
+	return methodSpecificId, methodSpecificIdCondition, nil
+}
+
+// getMSINonBlockchainAccountIdCondition asserts if the Method Specific Id is a CAIP-10 Blockchain Account Id
+func getMSINonBlockchainAccountIdCondition(didId string) bool {
+	didIdElements := strings.Split(didId, ":")
+	return (len(didIdElements) == 3 || len(didIdElements) == 4)
+}
+
+// getMSIMultibaseCondition asserts if the Method Specific Id is a Multibase encoded string
```
Comment (Vishwas1) [inline x/ssi/types/utils.go:126]: update the comment

File: x/ssi/types/utils.go [hunk 2]
```diff
@@ -46,3 +47,84 @@ func FindInSlice(list []string, element string) bool {
...
+func getMSIBlockchainAccountIdCondition(didId string) bool {
```
Comment (Vishwas1) [inline x/ssi/types/utils.go:127]: better change the variable name to reflect its a condition

**Review by Vishwas1 — CHANGES_REQUESTED** [review]
Please update

File: x/ssi/types/utils.go (hunk 1 above)
Comment (arnabghose997) [reply x/ssi/types/utils.go]: @Vishwas1  Renamed the following:

- `getMSIBlockchainAccountIdCondition` to `isMSIBlockchainAccountId`
- `getMSINonBlockchainAccountIdCondition` to `isMSINonBlockchainAccountId`

File: x/ssi/types/utils.go (hunk 2 above)
Comment (arnabghose997) [reply x/ssi/types/utils.go]: @Vishwas1  Renamed the following:

- `getMSIBlockchainAccountIdCondition` to `isMSIBlockchainAccountId`
- `getMSINonBlockchainAccountIdCondition` to `isMSINonBlockchainAccountId`

File: x/ssi/types/utils.go (hunk 1 above)
Comment (arnabghose997) [reply x/ssi/types/utils.go]: fixed the comment

**Review by Vishwas1 — APPROVED** [review]

**Pull request merged**